HinterStrategy: JavaDoc, More Preferences

rukemn opened the pull request:
Hint rating can now be tuned via Preference
Whether Hint or Treasure is picked first via Preference
Javadocs

Fixes #234
Fixes #284

rukemn committed:
added SteadyDirection Searcher

rukemn committed:
Fixed a HideAndSeekBug
Naive Anglesearcher is now also a HideAndSeekSearcher

rukemn committed:
Merge remote-tracking branch 'origin/master' into Hinterstrategy

rukemn committed:
Merge remote-tracking branch 'origin/SimpleSearchers' into Hinterstrategy

rukemn committed:
Some Colors, and sampling of the Worst Constant Point.

rukemn committed:
implemented the Calculation of WorstConstantPoint
Bugfix for integrateHint(AngleHint)

rukemn committed:
rewrite of hintIntegration
inclusion of calcWorstConstantPoint
colors

rukemn committed:
Added normal Angle in GeometryAngle
Factored out HintEvaluation and Stats for the Angles

rukemn committed:
cleanup,
MinDistance of Treasure Position possible

rukemn committed:
Merge remote-tracking branch 'origin/master' into Hinterstrategy

# Conflicts were in

#	GeometryType
#	MaxAreaAngularHintStrategy
#	NaiveAngleSearcher
#   HideAndSeekEngine

rukemn committed:
bit of clean up

rukemn committed:
CI javadoc fix

rukemn committed:
decreased LOG level from "Debug" to "Trace"

rukemn committed:
first refactoring steps

rukemn committed:
Fixed Treasure Hide refactoring done, still Test and Evaluation toDo

rukemn committed:
Update SearchPath.java

rukemn committed:
Update NaiveAngleSearcher.java

rukemn committed:
added Status messages, centroid evaluation

rukemn committed:
User ControlledSearcher Searchpath fix

rukemn committed:
bugfix; adaptBoundingCircle() wasn't correctly reintegrating the hints

rukemn committed:
MobileTreasureHider implementing HideAndSeek added

rukemn committed:
Merge remote-tracking branch 'origin/master' into Hinterstrategy

rukemn committed:
Some Preferences added

rukemn committed:
merging and codestyle

rukemn committed:
made SteadyDirection Searcher a HideAndSeekSearcher

rukemn committed:
fixed GeometryCollection bug,
added preference MaxCircleExtension

rukemn committed:
fixed bug in StaticHider

rukemn committed:
Merge remote-tracking branch 'origin/master' into Hinterstrategy

rukemn committed:
fixed Main-Preference issues

rukemn committed:
fixed a dumb bug

rukemn committed:
javadoc fix

rukemn committed:
added more javadoc
the rating weights are now a preference
MobileHider can now set whether the treasure or the hint should be picked first as preference

ghost commented:
## DeepCode's analysis on [#156e87](https://www.deepcode.ai/app/gh/jotoh98/treasureHunt/5247920079cff03d1328eb19d761d72bc7c3c079/jotoh98/treasureHunt/156e87ab18f4ebc279ca94ce651ae621ccfcd1c9/_/%2F/code/?utm_source=gh_review) found:

**3** minor issues.  :heavy_check_mark: **1** issue were fixed.

 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/jotoh98/treasureHunt/5247920079cff03d1328eb19d761d72bc7c3c079/jotoh98/treasureHunt/156e87ab18f4ebc279ca94ce651ae621ccfcd1c9/_/%2F/code/?utm_source=gh_review) 
---
:relaxed: If you want to provide feedback on our bot, here is how to [contact us](https://www.deepcode.ai/feedback?select=4&utm_source=gh_review).

rukemn committed:
added Halfplane Hint adapters for Statistical Hider subclasses

DorianRec commented on src/main/java/com/treasure/hunt/jts/geom/GeometryAngle.java:
```diff
@@ -63,6 +64,17 @@ public GeometryAngle(GeometryFactory factory, Coordinate center, double start, d
+     */
+    public double getNormalizedAngle(){
+        double angle = Angle.angle(this.getCenter() , JTSUtils.middleOfGeometryAngle(this));
+        if( angle < 0){
```
Please use autoformatting, which prevents **many** merge conflicts in later merges.
This could be used by pressing Strg+Alt+l for one file.
Better tick all these: 
<img width="679" alt="Bildschirmfoto 2020-03-29 um 18 13 44" src="https://user-images.githubusercontent.com/14947840/77854215-2cbb1c00-71e9-11ea-9994-1456c5e8e736.png">

rukemn replied:
Formatted

DorianRec reviewed (changes requested):
Simulating this with NaiveAngleSearcher, MobileTreasureHider and GameEngine, i got the following (Seems that the MobileTreasureHider gives Hints, which not includes the treasure, look at the last lines of stacktraces!):

18:24:50.370 [JavaFX Application Thread] INFO  c.t.h.s.h.impl.MobileTreasureHider -  --- MOVE STARTED ---
18:24:50.372 [JavaFX Application Thread] DEBUG c.t.h.s.h.impl.MobileTreasureHider - treasure first, hint second
18:24:50.374 [JavaFX Application Thread] DEBUG c.t.h.strategy.hider.impl.GameField - best overall calculated constant is0.6585091911355823 at (99.351255546, -10.247478613, NaN)
18:24:50.374 [JavaFX Application Thread] INFO  c.t.h.s.h.impl.MobileTreasureHider - trying to move the Treasure to POINT (99.351255546 -10.247478613)
18:24:50.375 [JavaFX Application Thread] INFO  c.t.h.strategy.hider.impl.GameField - new treasure location atPOINT (99.351255546 -10.247478613)
18:24:50.375 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - generating 360 hints of size 180.0
18:24:50.377 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - current possibleArea size 1963.1512533607531
18:24:50.451 [JavaFX Application Thread] DEBUG c.t.h.s.hider.impl.StatisticalHider - remaining possible Hints180
18:24:50.451 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - angleHint with the best rating
18:24:50.451 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - Hint at (35.31232982088212, 4.746345622951012, NaN) with normal angle of6.000000000000123, absolute Area Cutoff=16.34699123250425, relative Area Cutoff= 0.008326913784416535, Centroid distance to Treasure = 33.48392993566115, NormalLine distance to Treasure = 21.605576993314855, WorstPoint is =(0.0, 0.0, NaN) with C= 0.0}
18:24:50.451 [JavaFX Application Thread] INFO  c.t.h.s.h.impl.MobileTreasureHider -  --- MOVE ENDED ---
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1787)
	at javafx.fxml/javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1670)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Node.fireEvent(Node.java:8865)
	at javafx.controls/javafx.scene.control.Button.fire(Button.java:200)
	at javafx.controls/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:206)
	at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3876)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.access$1300(Scene.java:3604)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1874)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2613)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:397)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:434)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:395)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:433)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:556)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:942)
	at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor23.invoke(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.javafx.reflect.Trampoline.invoke(MethodUtil.java:76)
	at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at javafx.base/com.sun.javafx.reflect.MethodUtil.invoke(MethodUtil.java:275)
	at javafx.fxml/com.sun.javafx.fxml.MethodHelper.invoke(MethodHelper.java:83)
	at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1782)
	... 45 more
Caused by: java.lang.IllegalArgumentException: Treasure does not lie in given Angle.
	at com.treasure.hunt.game.GameEngine.verifyHint(GameEngine.java:155)
	at com.treasure.hunt.game.GameEngine.hiderMove(GameEngine.java:125)
	at com.treasure.hunt.game.GameEngine.move(GameEngine.java:113)
Caused by: java.lang.IllegalArgumentException: Treasure does not lie in given Angle.

	at com.treasure.hunt.game.GameManager.next(GameManager.java:168)
	at com.treasure.hunt.view.widget.NavigationController.nextButtonClicked(NavigationController.java:55)
	... 55 more

rukemn committed:
pretty up some visuals, some more work on adapters

rukemn commented:
"Treasure does not lie in given Angle. " 

This is not a bug caused by the hider, its from the GameEngine!

The GameEngine doesn#t check against the current target, but the initial one i think

rukemn committed:
formatting for Dorian

DorianRec reviewed (changes requested):
Unfortunately, I got an Exception, simulating the NaiveAngleHintSearcher, FixedTreasureSearcher and GameEngine:

21:43:03.825 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - generating 360 hints of size 180.0
21:43:03.827 [JavaFX Application Thread] INFO  c.t.h.s.hider.impl.StatisticalHider - current possibleArea size 6366.047884012224
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1787)
	at javafx.fxml/javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1670)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Node.fireEvent(Node.java:8865)
	at javafx.controls/javafx.scene.control.Button.fire(Button.java:200)
	at javafx.controls/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:206)
	at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3876)
	at javafx.graphics/javafx.scene.Scene$MouseHandler.access$1300(Scene.java:3604)
	at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1874)
	at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2613)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:397)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:434)
	at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:395)
	at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:433)
	at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:556)
	at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:942)
	at javafx.graphics/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Caused by: java.lang.reflect.InvocationTargetException

	at com.sun.javafx.reflect.Trampoline.invoke(MethodUtil.java:76)
	at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at javafx.base/com.sun.javafx.reflect.MethodUtil.invoke(MethodUtil.java:275)
	at javafx.fxml/com.sun.javafx.fxml.MethodHelper.invoke(MethodHelper.java:83)
	at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1782)
	... 45 more
Caused by: java.lang.IllegalArgumentException: Operation does not support GeometryCollection arguments
	at org.locationtech.jts.geom.Geometry.checkNotGeometryCollection(Geometry.java:1803)
	at org.locationtech.jts.geom.Geometry.union(Geometry.java:1386)
	at com.treasure.hunt.strategy.hider.impl.GameField.testHint(GameField.java:261)
Caused by: java.lang.IllegalArgumentException: Operation does not support GeometryCollection arguments

	at com.treasure.hunt.strategy.hider.impl.StatisticalHider.eval(StatisticalHider.java:96)
	at com.treasure.hunt.strategy.hider.impl.StatisticalHider.move(StatisticalHider.java:75)
	at com.treasure.hunt.strategy.hider.impl.FixedTreasureHider.move(FixedTreasureHider.java:13)
	at com.treasure.hunt.game.GameEngine.hiderMove(GameEngine.java:123)
	at com.treasure.hunt.game.GameEngine.move(GameEngine.java:113)
	at com.treasure.hunt.game.GameManager.next(GameManager.java:168)
	at com.treasure.hunt.view.widget.NavigationController.nextButtonClicked(NavigationController.java:55)
	... 55 more

DorianRec reviewed (changes requested):
Simulating the StrategyFromPaper versous HalfPlaneAdapter_FixedTreasure, threw me the following on the 116. move, which could also be an error in the StrategyFromPaper.

Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
	at com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.LastHintBadSubroutine.lastHintBadSubroutine(LastHintBadSubroutine.java:356)
	at com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.StrategyFromPaper.move(StrategyFromPaper.java:114)
	at com.treasure.hunt.strategy.searcher.impl.strategyFromPaper.StrategyFromPaper.move(StrategyFromPaper.java:33)
	at com.treasure.hunt.game.GameEngine.searcherMove(GameEngine.java:136)
	at com.treasure.hunt.game.GameEngine.move(GameEngine.java:107)
	at com.treasure.hunt.game.GameManager.next(GameManager.java:168)
	at com.treasure.hunt.game.GameManager.lambda$beat$10(GameManager.java:262)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$9(PlatformImpl.java:418)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:417)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)

rukemn committed:
bugfix for Issue#284

rukemn commented:
Fixed Issue 284, 
BadHintsubroutine is not caused by Statistical Hider I think

rukemn committed:
Improved treasure moving formula, option to use new or old formula;
added preference to move treasure only to positions, which are at least x distance away

rukemn committed:
refactor of MindDistancePreference; now in GameField instead of moblieTreasure

rukemn committed:
Merge remote-tracking branch 'origin/master' into Hinterstrategy

DorianRec reviewed (approved):
No errors found :)
